Application/Business/AssetType/AssetTypeServices.cs

```diff
@@ -30,14 +31,30 @@ public async Task<int> Count()
     }
     public async Task Create(AssetTypeEntity assetType)
     {
-
+        await CheckForDuplicateAssetType(assetType);
+        await _assetTypeRepo.CreateAsync(assetType);
     }
     public async Task Update(Guid assetTypeId, AssetTypeEntity assetType)
     {
-
+        await _assetTypeRepo.FoundOrThrowAsync(assetTypeId, "Asset type is not exist.");
+        await CheckForDuplicateAssetType(assetType);
+        await _assetTypeRepo.UpdateAsync(assetType);
     }
     public async Task Delete(Guid assetTypeId)
     {
-
+        await _assetTypeRepo.FoundOrThrowAsync(assetTypeId, "Asset type is not exist.");
+        await _assetTypeRepo.DeleteSoftAsync(assetTypeId);
+    }
+    public async Task CheckForDuplicateAssetType(AssetTypeEntity assetType)
+    {
+        var checkAssetType = await _assetTypeRepo.FirstOrDefaultAsync(
+            aT => aT.Name.Equals(assetType.Name) && aT.GameId.Equals(assetType.GameId));
+        if(checkAssetType is not null)
```

Comment:
this nested if can reduce by using where clause and check list.Count > 1.
FirstOrDefault returns the first element that satisfies the query, but what if you always get the element that satisfies the if clause but in the database there exist other elements that do not satisfy?